### PR TITLE
Booking Limits checking ALL events not the event the limit is applied too

### PR DESCRIPTION
**Pull request by sean-brydon**
This PR ensure we only check the event type the user is booking and not all bookings of OUR bookings in the users calendar.

**_Should_** fix the issue where we were getting an inconsistent result between availability being hidden on the calendar and the booking error message 

Translate error string on booking page if any. If there is no translation found we will just display the English error as per the normal usage of `useLocale`

CleanShot 2022-11-21 at 11 42 49@2x

### CODE REVIEW

File: packages/core/getUserAvailability.ts
Comment (sean-brydon) [inline packages/core/getUserAvailability.ts:207]: Only check OUR booking that matches the current eventTypeId - we don't care about another event type in this case

**Comment by emrysal** [comment]
Looks good to me - waiting for feedback before approval on the following:

💡 All the Github comments are probably suitable code comments for the code

File: packages/lib/server/checkBookingLimits.ts
Comment (emrysal) [inline packages/lib/server/checkBookingLimits.ts:89]: This should be translatable - this is meant for the FE but as it is meant for display it needs a translation string. We should name this error so we can use the name as a translation key (I think that's the easiest approach)

Comment (sean-brydon) [reply packages/lib/server/checkBookingLimits.ts:89]: How would we handle passing in the limitingNumber and filter to the FE error?

Comment (emrysal) [reply packages/lib/server/checkBookingLimits.ts]: For now let's keep it simple, we can come up with a more global strategy in the future as we fix similar HttpError's

**Review by emrysal — APPROVED** [review]

**Review by emrysal — APPROVED** [review]

**Pull request merged**
